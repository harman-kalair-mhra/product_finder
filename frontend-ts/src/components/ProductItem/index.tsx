import React, { useState, FormEvent } from "react";
import IProduct from "../../models/product";
import styled from "styled-components";

const StyledProductItem = styled.div`
  color: green;
  font-size: 18px;
  background-color: #f5f5dc;
  font-weight: 900
  border-style: solid;
  margin: 10px;
  padding: 10px;
`;

const Styledbutton = styled.button`
  background-color: #4caf50; /* Green */
  border: none;
  color: white;
  padding: 10px 22px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 2px;
  cursor: pointer;
  box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24),0 17px 50px 0 rgba(0,0,0,0.19);9);
  border-radius: 8px
`;

interface IProductItem {
  item: IProduct;
}

const ProductItem: React.FC<IProductItem> = (props) => {
  function handleclick() {
    console.log("It's Working");
    const requestOptions = {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        plNumber: props.item.plNumber,
        name: props.item.name,
        dose: props.item.dose,
        substance: props.item.substance,
        customerId: "c9c5b059-4afd-4201-a478-5a36bd2fbc61",
      }),
    };
    fetch("/order", requestOptions);
  }
  return (
    <StyledProductItem>
      <p>PLNumber: {props.item.plNumber}</p>
      <p>Name: {props.item.name}</p>
      <p>Dose: {props.item.dose}</p>
      <p>Substance: {props.item.substance}</p>
      <a href={props.item.url}>CLICK FOR INFORMATION</a>
      <br />
      <Styledbutton type="button" onClick={handleclick}>
        Select
      </Styledbutton>
    </StyledProductItem>
  );
};

export default ProductItem;
