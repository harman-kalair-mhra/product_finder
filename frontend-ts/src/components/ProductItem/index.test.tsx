import React from "react";
import { shallow, mount } from "enzyme";

import ProductItem from "./index";

let item = {
  plNumber: "PL123456",
  name: "Paracetamol",
  dose: "500mg",
  substance: "Paracetamol",
  url: "www.drug.com",
};

describe("Product Item", () => {
  it("renders shallow", () => {
    const component = shallow(<ProductItem item={item} />);
    expect(component).toMatchSnapshot();
  }),
    it("renders mount", () => {
      const component = mount(<ProductItem item={item} />);
      expect(component).toMatchSnapshot();
    });
});
