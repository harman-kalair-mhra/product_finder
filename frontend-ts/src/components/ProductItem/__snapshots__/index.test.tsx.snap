// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Item renders mount 1`] = `
<ProductItem
  item={
    Object {
      "dose": "500mg",
      "name": "Paracetamol",
      "plNumber": "PL123456",
      "substance": "Paracetamol",
      "url": "www.drug.com",
    }
  }
>
  <styled.div>
    <div
      className="sc-AxjAm cJxcDN"
    >
      <p>
        PLNumber: 
        PL123456
      </p>
      <p>
        Name: 
        Paracetamol
      </p>
      <p>
        Dose: 
        500mg
      </p>
      <p>
        Substance: 
        Paracetamol
      </p>
      <a
        href="www.drug.com"
      >
        CLICK FOR INFORMATION
      </a>
      <br />
      <styled.button
        onClick={[Function]}
        type="button"
      >
        <button
          className="sc-AxirZ khuxrF"
          onClick={[Function]}
          type="button"
        >
          Select
        </button>
      </styled.button>
    </div>
  </styled.div>
</ProductItem>
`;

exports[`Product Item renders shallow 1`] = `
<styled.div>
  <p>
    PLNumber: 
    PL123456
  </p>
  <p>
    Name: 
    Paracetamol
  </p>
  <p>
    Dose: 
    500mg
  </p>
  <p>
    Substance: 
    Paracetamol
  </p>
  <a
    href="www.drug.com"
  >
    CLICK FOR INFORMATION
  </a>
  <br />
  <styled.button
    onClick={[Function]}
    type="button"
  >
    Select
  </styled.button>
</styled.div>
`;

exports[`Product Select renders mount 1`] = `
<ProductItem
  item={
    Object {
      "dose": "500mg",
      "name": "Paracetamol",
      "plNumber": "PL123456",
      "substance": "Paracetamol",
      "url": "www.drug.com",
    }
  }
>
  <styled.div>
    <div
      className="sc-AxjAm cJxcDN"
    >
      <p>
        PLNumber: 
        PL123456
      </p>
      <p>
        Name: 
        Paracetamol
      </p>
      <p>
        Dose: 
        500mg
      </p>
      <p>
        Substance: 
        Paracetamol
      </p>
      <a
        href="www.drug.com"
      >
        CLICK FOR INFORMATION
      </a>
      <br />
      <styled.button
        onClick={[Function]}
        type="button"
      >
        <button
          className="sc-AxirZ khuxrF"
          onClick={[Function]}
          type="button"
        >
          Select
        </button>
      </styled.button>
    </div>
  </styled.div>
</ProductItem>
`;

exports[`Product Select renders shallow 1`] = `
<styled.div>
  <p>
    PLNumber: 
    PL123456
  </p>
  <p>
    Name: 
    Paracetamol
  </p>
  <p>
    Dose: 
    500mg
  </p>
  <p>
    Substance: 
    Paracetamol
  </p>
  <a
    href="www.drug.com"
  >
    CLICK FOR INFORMATION
  </a>
  <br />
  <styled.button
    onClick={[Function]}
    type="button"
  >
    Select
  </styled.button>
</styled.div>
`;
